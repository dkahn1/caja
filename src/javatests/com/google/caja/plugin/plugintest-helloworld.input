=== test:///helloworld.gxp
<gxp:template name="sayHello">
  <h1>Hello World</h1>
</gxp:template>

=== test:///helloworld.css
h1 { color: black; background: white }

=== test:///helloworld.js
function main() {
  document.getElementById('hello-base').innerHTML = sayHello();
}
