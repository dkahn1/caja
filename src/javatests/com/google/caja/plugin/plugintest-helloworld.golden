.pre h1 {
  color: black;
  background: white
}

{
  ___.loadModule(function (___OUTERS___) {
      ___OUTERS___.main = ___.simpleFunc(function main() {
          (function () {
              var x___ = ___.asSimpleFunc(___OUTERS___.getElementById)('hello-base');
              var x0___ = ___.asSimpleFunc(___.primFreeze(___OUTERS___.sayHello))();
              return x___.setInnerHTML_canCall___ ? x___.setInnerHTML(x0___) : ___.callPub(x___, 'setInnerHTML', [x0___]);
            })();
        });
      ___OUTERS___.sayHello = ___.simpleFunc(function sayHello() {
          var out___ = [];
          out___.push('<h1>Hello World</h1>');
          return ___OUTERS___.plugin_blessHtml___(out___.join(''));
        });
      var c1___ = ___.primFreeze(___OUTERS___.sayHello);
    });
}
